### PR TITLE
More code inspections

### PR DESCRIPTION
**Pull request by ufleisch**

### CODE REVIEW

**Review by neheb — APPROVED** [review]

File: taglib/ogg/opus/opusproperties.cpp [hunk 1]
```diff
@@ -138,9 +138,7 @@ void Opus::Properties::read(File *file)
     const long long end   = last->absoluteGranularPosition();
 
     if(start >= 0 && end >= 0) {
-      const long long frameCount = (end - start - preSkip);
-
-      if(frameCount > 0) {
+      if(const long long frameCount = (end - start - preSkip); frameCount > 0) {
```
Comment (neheb) [inline taglib/ogg/opus/opusproperties.cpp]: No need for the second part.

File: taglib/ogg/speex/speexproperties.cpp [hunk 2]
```diff
@@ -154,9 +154,7 @@ void Speex::Properties::read(File *file)
     const long long end   = last->absoluteGranularPosition();
 
     if(start >= 0 && end >= 0 && d->sampleRate > 0) {
-      const long long frameCount = end - start;
-
-      if(frameCount > 0) {
+      if(const long long frameCount = end - start; frameCount > 0) {
```
Comment (neheb) [inline taglib/ogg/speex/speexproperties.cpp:157]: No need for the second part.

File: taglib/ogg/vorbis/vorbisproperties.cpp [hunk 3]
```diff
@@ -157,9 +157,7 @@ void Vorbis::Properties::read(File *file)
     const long long end   = last->absoluteGranularPosition();
 
     if(start >= 0 && end >= 0 && d->sampleRate > 0) {
-      const long long frameCount = end - start;
-
-      if(frameCount > 0) {
+      if(const long long frameCount = end - start; frameCount > 0) {
```
Comment (neheb) [inline taglib/ogg/vorbis/vorbisproperties.cpp:160]: No need for the second part.

File: taglib/riff/wav/wavproperties.cpp [hunk 4]
```diff
@@ -172,8 +171,7 @@ void RIFF::WAV::Properties::read(File *file)
     d->bitrate = static_cast<int>(streamLength * 8.0 / length + 0.5);
   }
   else {
-    const unsigned int byteRate = data.toUInt(8, false);
-    if(byteRate > 0) {
+    if(const unsigned int byteRate = data.toUInt(8, false); byteRate > 0) {
```
Comment (neheb) [inline taglib/riff/wav/wavproperties.cpp:174]: No need for the second part.

File: taglib/mpeg/mpegproperties.cpp [hunk 5]
```diff
@@ -229,15 +229,15 @@ void MPEG::Properties::read(File *file, ReadStyle readStyle)
...
+         lastFrameOffset < 0) {
         debug("MPEG::Properties::read() -- Could not find an MPEG frame in the stream.");
       }
       else
       {
         const Header lastHeader(file, lastFrameOffset, false);
-        const offset_t streamLength = lastFrameOffset - firstFrameOffset + lastHeader.frameLength();
-        if (streamLength > 0)
+        if(const offset_t streamLength = lastFrameOffset - firstFrameOffset + lastHeader.frameLength();
+           streamLength > 0)
```
Comment (neheb) [inline taglib/mpeg/mpegproperties.cpp:239]: No need for this.

File: taglib/ogg/oggfile.cpp [hunk 6]
```diff
@@ -269,10 +269,9 @@ void Ogg::File::writePacket(unsigned int i, const ByteVector &packet)
 
   // Renumber the following pages if the pages have been split or merged.
 
-  const int numberOfNewPages
-    = pages.back()->pageSequenceNumber() - lastPage->pageSequenceNumber();
-
-  if(numberOfNewPages != 0) {
+  if(const int numberOfNewPages
+      = pages.back()->pageSequenceNumber() - lastPage->pageSequenceNumber();
+     numberOfNewPages != 0) {
```
Comment (neheb) [inline taglib/ogg/oggfile.cpp:274]: If this can't be negative, it can be removed.

File: tests/test_apetag.cpp [hunk 7]
```diff
@@ -89,10 +89,10 @@ class TestAPETag : public CppUnit::TestFixture
   void testPropertyInterface2()
   {
     APE::Tag tag;
-    APE::Item item1 = APE::Item("TRACK", String("17"));
+    auto item1 = APE::Item("TRACK", String("17"));
```
Comment (neheb) [inline tests/test_apetag.cpp:92]: wonder if it makes more sense to do

```
APE::Item item1("TRACK", String("17");
```

to look more like APE::Tag tag;

Comment (ufleisch) [reply tests/test_apetag.cpp:92]: OK, I will change this and the other similar items.

Comment (ufleisch) [reply tests/test_apetag.cpp:92]: All except one of these "No need for the second part" concern _signed_ variables, so they could be negative with specially crafted or fuzzed files, and leaving the second part would let such negative values pass, so I would rather leave it as it is. Or am I missing something?

Comment (neheb) [reply tests/test_apetag.cpp]: AFAIK if statements are booleans. Negative values evaluate to 0.

Comment (ufleisch) [reply tests/test_apetag.cpp]: No, only zero is falsy, see [Boolean Conversions](https://en.cppreference.com/w/cpp/language/implicit_conversion#Boolean_conversions) or try it out [online](https://www.w3schools.com/cpp/trycpp.asp?filename=demo_compiler).

**Comment by sbooth** [comment]
@ufleisch Your efforts getting TagLib ready for 2.0 are quite impressive and much appreciated!

**Pull request merged**